File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& arr) {
        int n = arr.size();
        if( n < 1)return 0;
        sort(arr.begin()  , arr.end());
        int cnt = 1 , maxi = 1 , ele = arr[0];
        for(int i = 1 ; i < n ; i++){
            if(arr[i] == arr[i-1])continue;
            if(arr[i] == ele + 1 )cnt++;
            else{
                maxi = max(maxi , cnt);
                cnt = 1;
            }
            ele = arr[i];
        }
        return max(maxi , cnt);
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
    //  if(nums.isEmpty())return 0;
        Arrays.sort(nums);
        int ans = 1 , count = 1 , n = nums.length;
        if(n < 1)return 0;
        int start = nums[0];
        for(int i = 1 ; i < n ; i++){
            if(nums[i-1] == nums[i])continue;
            if(start + 1 == nums[i])count++;
            else {
                ans = Math.max(ans , count);
                count = 1 ;
            }
            start = nums[i];


        }
        return Math.max(ans , count);       
    }
}